### PR TITLE
Cleanup mime attribute of enableaudiodevicecallback

### PR DESCRIPTION
**Pull request by borongc**
Remove this attribute as it is not used.

b/216483296

### CODE REVIEW

File: starboard/android/shared/player_components_factory.h [hunk 1]
```diff
@@ -215,13 +215,11 @@ class PlayerComponentsFactory : public starboard::shared::starboard::player::
...
       MimeType audio_mime_type(creation_parameters.audio_mime());
       if (!audio_mime_type.is_valid() ||
-          !audio_mime_type.ValidateBoolParameter("enableaudiodevicecallback") ||
           !audio_mime_type.ValidateBoolParameter("audiopassthrough")) {
         return scoped_ptr<PlayerComponents>();
       }
 
-      enable_audio_device_callback =
-          audio_mime_type.GetParamBoolValue("enableaudiodevicecallback", true);
+      enable_audio_device_callback = true;
```
Comment (xiaomings) [inline starboard/android/shared/player_components_factory.h:222]: We should remove all code related to variable `enable_audio_device_callback` and assume its value is true.

Comment (borongc) [reply starboard/android/shared/player_components_factory.h]: Cleanup enable_audio_device_callback and leave it as default value of true.

File: starboard/android/shared/player_components_factory.h [hunk 2]
```diff
@@ -97,7 +97,7 @@ class AudioRendererSinkAndroid : public ::starboard::shared::starboard::player::
                   audio_frame_storage_type, frame_buffers,
                   frame_buffers_size_in_frames, update_source_status_func,
                   consume_frames_func, error_func, start_media_time,
-                  tunnel_mode_audio_session_id, enable_audio_device_callback,
+                  tunnel_mode_audio_session_id, true,
```
Comment (osagie98) [inline starboard/android/shared/player_components_factory.h:100]: You can go further and remove this parameter entirely. This makes the change a little more complex as you'll need to modify at least audio_track_audio_sink_type.* and AudioOutputManager.java, but the end result will be simpler.

Comment (borongc) [reply starboard/android/shared/player_components_factory.h]: Cleanup the parameter entirely.

File: starboard/android/shared/audio_track_audio_sink_type.cc [hunk 3]
```diff
@@ -433,16 +431,14 @@ SbAudioSink AudioTrackAudioSinkType::Create(
   const SbTime kStartTime = 0;
   // Disable tunnel mode.
   const int kTunnelModeAudioSessionId = -1;
-  // Disable AudioDeviceCallback for WebAudio.
```
Comment (osagie98) [inline starboard/android/shared/audio_track_audio_sink_type.cc:436]: We should still ensure that audioDeviceCallback is disabled for WebAudio.

File: starboard/android/shared/player_components_factory.h [hunk 4]
```diff
@@ -235,9 +226,9 @@ class PlayerComponentsFactory : public starboard::shared::starboard::player::
     SB_LOG(INFO) << "Creating passthrough components.";
     // TODO: Enable tunnel mode for passthrough
     scoped_ptr<AudioRendererPassthrough> audio_renderer;
-    audio_renderer.reset(new AudioRendererPassthrough(
-        creation_parameters.audio_stream_info(),
-        creation_parameters.drm_system(), enable_audio_device_callback));
+    audio_renderer.reset(
+        new AudioRendererPassthrough(creation_parameters.audio_stream_info(),
+                                     creation_parameters.drm_system(), true));
```
Comment (osagie98) [inline starboard/android/shared/player_components_factory.h:231]: You can remove this param from AudioRenderPassthrough as well

Comment (borongc) [reply starboard/android/shared/player_components_factory.h]: Fixed it.

File: starboard/android/shared/audio_track_audio_sink_type.cc (hunk 3 above)
Comment (borongc) [reply starboard/android/shared/audio_track_audio_sink_type.cc:436]: Fixed.

**Review by osagie98 — APPROVED** [review]

**Pull request merged**